packages/editor-web-component/src/editor-web-component.tsx

```diff
@@ -0,0 +1,89 @@
+import { SerloEditor } from '@serlo/editor'
+import React from 'react'
+import * as ReactDOM from 'react-dom/client'
+
+import {
+  exampleInitialState,
+  isInitialState,
+  type InitialState,
+} from './initial-state'
+
+// Could probably remove the export entirely, as the customElement is registered
+// below.
+export class EditorWebComponent extends HTMLElement {
+  private reactRoot: ReactDOM.Root | null = null
+  private container: HTMLDivElement
+
+  constructor() {
+    super()
+
+    this.container = document.createElement('div')
+  }
+
+  attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    console.log('AttributeChangedCallback', {
+      name,
+      oldValue,
+      newValue,
+    })
+    if (oldValue !== newValue) {
+      this.mountReactComponent()
+    }
+  }
+
+  static get observedAttributes() {
+    return ['initial-state']
```

Comment:
this is not black or white but exposing structured data as an attribute seems weird to me. Internally we usually expose through a property (in lit syntax: `<my-el .initialState=${{key: 'value'}}></my-el>`) to avoid confusion around serialization. All webcomponent environments allow some sort of property binding since there's always things you won't be able to do with native DOM APIs. 

To allow for serializing the serlo editor invocation including its initial state, you could allow both a property and the attribute. You'd just need to `??` in ln49 and add the respective class member. 

To the argument "stay as close to DOM APIs as possible" while a good point in principle, I would reply that you're going to be forced to have some sort of more complex API very soon anyways when you want to have any kind of plugin adding API, especially if you're going to allow for custom plugins.